src/app/products/product-details/product-details.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, OnInit } from '@angular/core';
+import { Input } from '@angular/core';
+import { Product } from '../../models/product';
+import { ProductService } from '../product.service';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-product-details',
+  templateUrl: './product-details.component.html',
+  styleUrls: ['./product-details.component.scss'],
+})
+export class ProductDetailsComponent implements OnInit {
+  @Input()
+  product: Product;
+
+  constructor(
+    private productService: ProductService,
+    private route: ActivatedRoute
+  ) {}
+
+  ngOnInit(): void {
+    const id = this.route.snapshot.params['id'];
+    if (id == undefined) {
```

Comment:
1. ta sytuacja jest teoretycznie niemozliwa - jak nie bedzie id to nawet ci sie ten komponent nie zaincjalizuje :) i nie ma opcji zeby ci sie zainicjalizowal z undefined.
2. id == undefined  PODWÓJNE FUUUUUJ nie dosc ze uzywasz == ktorego nigdy przenigdy nie powinno sie uzywac  to dodatkowo ten warunek nie sprawdzi ci innych przypadkow ktore sie nie nadaja (pusty string,, false etc)

Reply:
wywalone